app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -176,4 +178,32 @@ def parse_image(image, is_public)
 
     return uid, new_result
   end
+
+  def parse_vapp_template(vapp_template)
+    uid = vapp_template.id
+
+    begin
+      # The content of the template is the OVF specification of the vApp template
+      content = @connection.get_vapp_template_ovf_descriptor(uid).body
+
+      new_result = {
+        :type        => ManageIQ::Providers::Vmware::CloudManager::OrchestrationTemplate.name,
+        :ems_ref     => uid,
+        :name        => vapp_template.name,
+        :description => vapp_template.description,
+        :orderable   => false,
+        :content     => content,
+        # By default #save_orchestration_templates_inventory does not set the EMS
+        # ID because templates are not EMS specific. We are setting the EMS
+        # explicitly here, because vapps are specific to concrete EMS.
+        :ems_id      => @ems.id
+      }
+
+      return uid, new_result
+    rescue
+      # Without content we should not create the orchestration template because
+      # these are identified by it (or its MD5)
+      $log.info("MIQ(#{self.class.name}.#{__method__}) Downloading vApp template for vApp template #{uid} failed")
```

Comment:
I think the most important question is what we'd like to do with these templates. I haven't seen where they are currently used so it is not easy for me to decide what would be the best way to proceed.

Definitely, if we set the content to it's ID, we remove this download issue entirely, but definitelly introduce a potential issue in the future due to the fact that the content column is not (semantically) properly filled in.

We should keep OVF download if it would make sense to migrate vApp templates between EMSes in the future. If we have the template and it's OVF, we should be able to (theoretically) recreate the template for the other EMS - with some limitations to the network and storage profiles probably.

If we go with the second one, then I wonder if I need to do any error handling at all? If the exception occurs during OVF download, it should be propagated all the way to the EmsRefresh where it is properly logged and the refresh is abandoned. I may still add a log entry and raise another exception if the download fails?